presto-main/src/main/java/io/prestosql/sql/planner/LogicalPlanner.java

```diff
@@ -374,8 +373,15 @@ private RelationPlan createInsertPlan(Analysis analysis, Insert insertStatement)
                     Expression cast = noTruncationCast(input.toSymbolReference(), queryType, tableType);
                     assignments.put(output, cast);
                 }
+                columnsToBeInserted.add(columns.get(column.getName()));
+                visibleColumnsBuilder.add(column);
             }
         }
+
+        List<ColumnMetadata> visibleTableColumns = visibleColumnsBuilder.build();
+        List<String> visibleTableColumnNames = visibleTableColumns.stream()
```

Comment:
It seems that `columnsToBeInserted` is equal to `visibleTableColumnNames`

Reply:
`columnsToBeInserted` has the `ColumnHandle` which we might need to pass for `beginInsert` operation so we might need it